### PR TITLE
Zubin Chopra - Settings

### PR DESCRIPTION
**Pull request by zubinchopra**
## Account Settings Front End
Front end code for the member accounts component for both admin and general members

### Changes made
* Added the component to work for all roles.
* Using data from account object obtained from firebase.

### Notes
* Backend code needs to be written to reflect the changes in the DB as well
* Add button does not work right now. Clicking on the button will give the text "Coming soon..." (Had a chat with Ariel about this)

### CODE REVIEW

File: src/PageContainer.js [hunk 1]
```diff
@@ -225,7 +226,10 @@ class PageContainer extends Component {
 
                 {this.state.content === PageContent.SETTINGS && (
                     <div style={{ marginTop: '120px', marginLeft: '250px' }}>
```
Comment (arielin3) [inline src/PageContainer.js]: move to css

File: src/PageContent/Settings/OrganizationDetails.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                </div>
+
+                {!this.state.isEditing ?
+
+                    <div className="scs-0-content">
+                        <h5>Organization Details</h5>
+                        <h6>{this.props.account.name}</h6>
+                        <div className="scs-spacing" />
+                        <h6>Address: {this.props.account.address.street1}, {this.props.account.address.street2 ? <span>{this.props.account.address.street2},</span> : <span />} {this.props.account.address.city}, {this.props.account.address.state} - {this.props.account.address.zipcode}</h6>
+                        {this.props.account.phone ? <h6>Phone: {this.props.account.organizationPhone}</h6> : <span />}
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: (did I add phone here in the sketch? lol) there are no phones for organization, so can remove

File: src/PageContent/Settings/OrganizationDetails.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                    </div>
+
+                    :
+
+                    <div className="scs-0-content scs-1-editing">
+                        <h5>Organization Details</h5>
+
+                        <form onSubmit={this.handleSubmit.bind(this)}>
+                            <div className="editing-box">
+                                <div className="editing-child-1">
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: The label columns and the input fields are a little unaligned (in the UI)

File: src/PageContent/Settings/OrganizationDetails.js [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                                    <label className="label-component details">Name</label><br /><br />
+                                    <div className="scs-spacing" />
+                                    <div className="scs-spacing" />
+                                    <label className="label-component details">Street 1</label><br /><br />
+                                    <label className="label-component details">Street 2</label><br /><br />
+                                    <label className="label-component details">City</label><br /><br />
+                                    <label className="label-component details">State</label><br /><br />
+                                    <label className="label-component details">Zip</label><br /><br />
+                                    <div className="scs-spacing" />
+                                    <label className="label-component details">Phone</label><br /><br />
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: remove phone

File: src/PageContent/Settings/OrganizationDetails.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+                    />
+                </div>
+
+                {!this.state.isEditing ?
+
+                    <div className="scs-0-content">
+                        <h5>Organization Details</h5>
+                        <h6>{this.props.account.name}</h6>
+                        <div className="scs-spacing" />
+                        <h6>Address: {this.props.account.address.street1}, {this.props.account.address.street2 ? <span>{this.props.account.address.street2},</span> : <span />} {this.props.account.address.city}, {this.props.account.address.state} - {this.props.account.address.zipcode}</h6>
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: add `officeNo` (which is part of the address, if they entered one)
```
{this.props.account.address.officeNo &&
    <h6>Office Number: {this.props.account.address.officeNo}</h6>
```

File: src/PageContent/Settings/OrganizationDetails.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+                            <div className="editing-box">
+                                <div className="editing-child-1">
+                                    <label className="label-component details">Name</label><br /><br />
+                                    <div className="scs-spacing" />
+                                    <div className="scs-spacing" />
+                                    <label className="label-component details">Street 1</label><br /><br />
+                                    <label className="label-component details">Street 2</label><br /><br />
+                                    <label className="label-component details">City</label><br /><br />
+                                    <label className="label-component details">State</label><br /><br />
+                                    <label className="label-component details">Zip</label><br /><br />
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: add `Office Number`

File: src/PageContent/Settings/OrganizationDetails.js [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+                {!this.state.isEditing ?
+
+                    <div className="scs-0-content">
+                        <h5>Organization Details</h5>
+                        <h6>{this.props.account.name}</h6>
+                        <div className="scs-spacing" />
+                        <h6>Address: {this.props.account.address.street1}, {this.props.account.address.street2 ? <span>{this.props.account.address.street2},</span> : <span />} {this.props.account.address.city}, {this.props.account.address.state} - {this.props.account.address.zipcode}</h6>
+                        {this.props.account.phone ? <h6>Phone: {this.props.account.organizationPhone}</h6> : <span />}
+                        {this.props.account.numVolunteers ? <h6>Volunteers: {this.props.account.numVolunteers}</h6> : <span />}
+                        {this.props.account.deliveryNotes ? <h6>Notes: {this.props.account.deliveryNotes}</h6> : <span />}
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: "Delivery note:"

File: src/PageContent/Settings/OrganizationDetails.js [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+                                    <div className="scs-spacing" />
+                                    <label className="label-component details">Phone</label><br /><br />
+                                    {this.props.account.numVolunteers ? 
+                                        <label className="label-component details">Volunteers</label>
+                                        :
+                                        <span />
+                                    }
+                                    {this.props.account.deliveryNotes ? 
+                                        <div>
+                                            <label className="label-component details">Notes</label><br /><br />
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: "Delivery note"

File: src/PageContent/Settings/OrganizationDetails.js [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+                                    <input name="name" type="text" className="form-input" defaultValue={this.props.account.name} /><br /><br />
+                                    <div className="scs-spacing" />
+                                    <input name="street1" type="text" className="form-input" defaultValue={this.props.account.address.street1} /><br /><br />
+                                    <input name="street2" type="text" className="form-input" defaultValue={this.props.account.address.street2} /><br /><br />
+                                    <input name="city" type="text" className="form-input" defaultValue={this.props.account.address.city} /><br /><br />
+                                    <input name="state" type="text" className="form-input" defaultValue={this.props.account.address.state} /><br /><br />
+                                    <input name="zip" type="text" className="form-input" defaultValue={this.props.account.address.zipcode} /><br /><br />
+                                    <input name="phone" type="tel" pattern={StringFormat.PHONE} className="form-input" defaultValue={this.props.account.organizationPhone} /><br /><br />
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: remove

File: src/PageContent/Settings/OrganizationDetails.js [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        //     notes = e.target.notes.value;
+        // if(e.target.ep)
+        //     ep = e.target.ep.checked;
+        // let name = e.target.name.value;
+        // let address = {
+        //     street1: e.target.street1.value,
+        //     street2: e.target.street2.value,
+        //     city: e.target.city.value,
+        //     state: e.target.state.value,
+        //     zipcode: e.target.zip.value
```
Comment (arielin3) [inline src/PageContent/Settings/OrganizationDetails.js]: \+ `officeNo`

File: src/PageContent/Settings/Settings.js [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+import OrganizationDetails from './OrganizationDetails';
+import AccountManager from './AccountManager';
+import { AccountType } from '../../Enums';
+import MemberAccount from './MemberAccount';
+import PersonalAccount from './PersonalAccount';
+import './SCSettings0.css';
+import './SCSettings1.css';
+
+class Settings extends Component {
+
```
Comment (arielin3) [inline src/PageContent/Settings/Settings.js]: we need to fetch for Donating Agency's primaryContact data for `AccountManager` section. Add `componentDidMount()`:
```
componentDidMount(): {
    let manager = this.props.account;
    if (manager.accountType === AccountType.DONATING_AGENCY_MEMBER && !manager.isAdmin) {
        // backend TODO: fetch DA primary contact data for AccountManager
        manager = {
            accountType: AccountType.DONATING_AGENCY_MEMBER,
            name: 'something',
            email: 'something',
            phone: 'something',
            position: 'something',
        };
    }
    this.setState({manager: manager});
}
```

File: src/PageContent/Settings/Settings.js [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+                                account={org}
+                            />
+                        </div>
+
+                        <div className="scs-spacing" />
+
+                        <div className="container">
+                            <AccountManager
+                                accountType={this.props.account}
+                                account={org}
```
Comment (arielin3) [inline src/PageContent/Settings/Settings.js]: don't need `accountType`, just `account={this.state.manager}` (see comment about adding `componentDidMount`)

File: src/PageContent/Settings/Settings.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+                        {this.props.account.accountType === AccountType.DONATING_AGENCY_MEMBER ?
+                            <div className="container">
+                                {this.props.account.isAdmin ?
+                                    <MemberAccount
+                                        account={org}
+                                    />
+                                    :
+                                    <PersonalAccount
+                                        account={org}
+                                    />
```
Comment (arielin3) [inline src/PageContent/Settings/Settings.js]: `account={this.props.account}`

File: src/PageContent/Settings/Settings.js [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+            org = this.props.donatingAgency;
+        return (
+            <div>
+                {this.props.account ?
+
+                    <div> 
+                        <div className="container">
+                            <OrganizationDetails
+                                accountType={this.props.account}
+                                account={org}
```
Comment (arielin3) [inline src/PageContent/Settings/Settings.js]: it's really confusing to have `accountType` and `account` the way you're using it. Maybe have 
```
daAccount={this.props.account}
org={org}
```
instead? so inside `OrganizationDetails` it's clear that one is `org` and the other one is just purely used for DA cases.

File: src/PageContent/Settings/AccountManager.js [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+                                    <div className="amd-details-child">
+                                        <h6>Recieve Notifications via</h6>
+                                        <h6>Feature coming soon!</h6>
+                                        {/* {this.props.account.smsNotif ? <h6>SMS/Text Message</h6> : <div/>}
+                                        {this.props.account.emailNotif ? <h6>Email</h6> : <div/>} */}
+                                    </div>
+                                </div>
+                            </div>
+
+                            {this.props.secondaryContact ?
```
Comment (arielin3) [inline src/PageContent/Settings/AccountManager.js]: `this.props.account.secondaryContact` same with all the below

File: src/PageContent/Settings/AccountManager.js [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+
+                                <div className="editing-child-3">
+                                    Recieve Notifications via<br />
+                                    <h6>Feature coming soon!</h6>
+                                    {/* <input type="checkbox" name="smsNotif" value="smsNotif" defaultChecked={this.props.account.smsNotif}/><label className="label-component details">SMS/Text Message</label><br />
+                                    <input type="checkbox" name="emailNotif" value="emailNotif" defaultChecked={this.props.account.emailNotif}/><label className="label-component details">Email</label> */}
+                                </div>    
+                            </div>
+
+                            {this.props.secondaryContact ?
```
Comment (arielin3) [inline src/PageContent/Settings/AccountManager.js]: `this.props.account.accountType === AccountType.RECEIVING_AGENCY ?`
we should allow them to add one if they didn't have one

File: src/PageContainer.js (hunk 1 above)
Comment (zubinchopra) [reply src/PageContainer.js]: Fixed.

File: src/PageContent/Settings/OrganizationDetails.js (hunk 7 above)
Comment (zubinchopra) [reply src/PageContent/Settings/OrganizationDetails.js]: Fixed.

File: src/PageContent/Settings/OrganizationDetails.js (hunk 8 above)
Comment (zubinchopra) [reply src/PageContent/Settings/OrganizationDetails.js]: Fixed.

File: src/PageContent/Settings/OrganizationDetails.js (hunk 9 above)
Comment (zubinchopra) [reply src/PageContent/Settings/OrganizationDetails.js]: Fixed.

Comment (zubinchopra) [reply src/PageContent/Settings/OrganizationDetails.js]: I have added the an input field for office number instead

File: src/PageContent/Settings/OrganizationDetails.js (hunk 10 above)
Comment (zubinchopra) [reply src/PageContent/Settings/OrganizationDetails.js]: Added

File: src/PageContent/Settings/Settings.js (hunk 11 above)
Comment (zubinchopra) [reply src/PageContent/Settings/Settings.js]: Added.

File: src/PageContent/Settings/Settings.js (hunk 12 above)
Comment (zubinchopra) [reply src/PageContent/Settings/Settings.js]: Changed.

File: src/PageContent/Settings/Settings.js (hunk 13 above)
Comment (zubinchopra) [reply src/PageContent/Settings/Settings.js]: Changed

File: src/PageContent/Settings/Settings.js (hunk 14 above)
Comment (zubinchopra) [reply src/PageContent/Settings/Settings.js]: Fixed.

File: src/PageContent/Settings/AccountManager.js (hunk 15 above)
Comment (zubinchopra) [reply src/PageContent/Settings/AccountManager.js]: Fixed.

**Pull request merged**